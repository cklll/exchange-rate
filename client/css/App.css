html, body {
    min-width:320px;
}

/* non-collapsing navbar
https://stackoverflow.com/questions/23535289/bootstrap-3-disable-navbar-collapse */
.navbar-collapse.collapse {
  display: block!important;
}
.navbar-nav>li, .navbar-nav {
  float: left !important;
}
.navbar-nav.navbar-right:last-child {
  margin-right: -15px !important;
}
.navbar-right {
  float: right!important;
}
.navbar-right {
    /* margin-top:5px;
    margin-bottom:5px; */
}

.converter {
    display: flex;
    min-height: 80vh;
    justify-content: center;
    flex-direction: column;
    text-align: left;
}
.converter .error {
    color: #a94442;
    font-style: italic;
    opacity: 0;
    position: absolute;
}
.converter .error.show {
    opacity: 1;
}
.converter .equal-to {
    text-align: center;
    margin-top: 10vh;
    margin-bottom: 10vh;
}

.flag {display: inline-block;width: 48px;height: 48px;background:url(../assets/flags.png);}
.flag.flag-us {
    background-position: -48px -48px;
}
.flag.flag-gb {
    background-position: -48px 0;
}
.flag.flag-hk {
    background-position: 0 -48px;
}
.flag.flag-eu {
    background-position: 0 0;
}
