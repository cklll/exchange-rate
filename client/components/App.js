import React, { Component } from 'react';

import '../css/App.css';
import Header from './Header';
import Converter from './Converter';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <div className="MainContainer">
          <Converter />
        </div>
      </div>
    );
  }
}

export default App;
