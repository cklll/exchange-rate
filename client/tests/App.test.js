
import React from 'react';
import App from '../components/App';
import renderer from 'react-test-renderer';

global.fetch = jest.fn().mockImplementation((url) => {
    var p = new Promise((resolve, reject) => {
      resolve({
        json: function() { 
            if (url == '/api/rates') {
                return {
                    base: 'USD',
                    rates: {
                        'USD': 1,
                        'GBP': 2,
                        'EUR': 3,
                        'HKD': 4,
                    }
                }
            } else {
                return {
                    "rates": {
                      "2018-05-28": {
                        "HKD": 7.84495,
                        "GBP": 0.751146
                      },
                      "2018-05-29": {
                        "HKD": 7.84575,
                        "GBP": 0.754477
                      },
                      "2018-05-30": {
                        "HKD": 7.84685,
                        "GBP": 0.752771
                      },
                      "2018-05-31": {
                        "HKD": 7.84625,
                        "GBP": 0.752503
                      },
                      "2018-06-01": {
                        "HKD": 7.84585,
                        "GBP": 0.748955
                      },
                      "2018-06-02": {
                        "HKD": 7.84575,
                        "GBP": 0.74912
                      },
                      "2018-06-03": {
                        "HKD": 7.84555,
                        "GBP": 0.748868
                      },
                      "2018-06-04": {
                        "HKD": 7.84605,
                        "GBP": 0.750977
                      },
                      "2018-06-05": {
                        "HKD": 7.8474,
                        "GBP": 0.745875
                      },
                      "2018-06-06": {
                        "HKD": 7.84745,
                        "GBP": 0.745045
                      },
                      "2018-06-07": {
                        "HKD": 7.84635,
                        "GBP": 0.745381
                      },
                      "2018-06-08": {
                        "HKD": 7.847138,
                        "GBP": 0.746
                      },
                      "2018-06-09": {
                        "HKD": 7.845938,
                        "GBP": 0.74563
                      },
                      "2018-06-10": {
                        "HKD": 7.84635,
                        "GBP": 0.7455
                      },
                      "2018-06-11": {
                        "HKD": 7.846383,
                        "GBP": 0.747863
                      },
                      "2018-06-12": {
                        "HKD": 7.84695,
                        "GBP": 0.747719
                      },
                      "2018-06-13": {
                        "HKD": 7.8478,
                        "GBP": 0.747253
                      },
                      "2018-06-14": {
                        "HKD": 7.84995,
                        "GBP": 0.754432
                      },
                      "2018-06-15": {
                        "HKD": 7.848868,
                        "GBP": 0.752845
                      },
                      "2018-06-16": {
                        "HKD": 7.848868,
                        "GBP": 0.752845
                      },
                      "2018-06-17": {
                        "HKD": 7.84935,
                        "GBP": 0.753263
                      },
                      "2018-06-18": {
                        "HKD": 7.84955,
                        "GBP": 0.754264
                      },
                      "2018-06-19": {
                        "HKD": 7.84935,
                        "GBP": 0.759108
                      },
                      "2018-06-20": {
                        "HKD": 7.84569,
                        "GBP": 0.759137
                      },
                      "2018-06-21": {
                        "HKD": 7.84575,
                        "GBP": 0.755013
                      },
                      "2018-06-22": {
                        "HKD": 7.84785,
                        "GBP": 0.75391
                      },
                      "2018-06-23": {
                        "HKD": 7.84785,
                        "GBP": 0.75391
                      },
                      "2018-06-24": {
                        "HKD": 7.846541,
                        "GBP": 0.754053
                      },
                      "2018-06-25": {
                        "HKD": 7.8467,
                        "GBP": 0.753137
                      },
                      "2018-06-26": {
                        "HKD": 7.849,
                        "GBP": 0.756395
                      },
                      "2018-06-27": {
                        "HKD": 7.848265,
                        "GBP": 0.762207
                      }
                    },
                    "base": "USD"
                  }
            }
        }
      });
    });

    return p;
});


test('App.js', () => {
  const component = renderer.create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});