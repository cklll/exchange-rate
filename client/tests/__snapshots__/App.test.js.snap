// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js 1`] = `
<div
  className="App"
>
  <header>
    <nav
      className="navbar navbar-default"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href=""
        >
          Currency Exchange Rate
        </a>
        <div
          className="collapse navbar-collapse"
          id="bs-example-navbar-collapse-1"
        >
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li>
              <a
                href="https://github.com/cklll/exchange-rate"
                target="_blank"
              >
                GitHub
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="MainContainer"
  >
    <div>
      <div
        className="converter container"
      >
        <form>
          <div
            className="container"
          >
            <div
              className="show-grid form-group row"
            >
              <div
                className="col-xs-12"
              >
                <div
                  className="dropdown btn-group btn-group-info"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="dropdown-toggle btn btn-info"
                    disabled={false}
                    id="fromDropwdown"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    type="button"
                  >
                    <span>
                      <img
                        alt="HKD"
                        className="flag flag-hk"
                        src="../assets/blank.gif"
                      />
                      HKD
                    </span>
                     
                    <span
                      className="caret"
                    />
                  </button>
                  <ul
                    aria-labelledby="fromDropwdown"
                    className="dropdown-menu"
                    role="menu"
                  >
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="USD"
                          className="flag flag-us"
                          src="../assets/blank.gif"
                        />
                        USD
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="HKD"
                          className="flag flag-hk"
                          src="../assets/blank.gif"
                        />
                        HKD
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="EUR"
                          className="flag flag-eu"
                          src="../assets/blank.gif"
                        />
                        EUR
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="GBP"
                          className="flag flag-gb"
                          src="../assets/blank.gif"
                        />
                        GBP
                      </a>
                    </li>
                  </ul>
                </div>
                <input
                  className="form-control"
                  defaultValue={1}
                  min="0"
                  onChange={[Function]}
                  type="number"
                />
                <p
                  className="error"
                >
                  Please input a positive number
                </p>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="equal-to col-xs-12"
              >
                <button
                  className="btn btn-primary btn-swap"
                  onClick={[Function]}
                  type="button"
                >
                  â‡…
                </button>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-12"
              >
                <div
                  className="dropdown btn-group btn-group-info"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="dropdown-toggle btn btn-info"
                    disabled={false}
                    id="toDropwdown"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    type="button"
                  >
                    <span>
                      <img
                        alt="USD"
                        className="flag flag-us"
                        src="../assets/blank.gif"
                      />
                      USD
                    </span>
                     
                    <span
                      className="caret"
                    />
                  </button>
                  <ul
                    aria-labelledby="toDropwdown"
                    className="dropdown-menu"
                    role="menu"
                  >
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="USD"
                          className="flag flag-us"
                          src="../assets/blank.gif"
                        />
                        USD
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="HKD"
                          className="flag flag-hk"
                          src="../assets/blank.gif"
                        />
                        HKD
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="EUR"
                          className="flag flag-eu"
                          src="../assets/blank.gif"
                        />
                        EUR
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="GBP"
                          className="flag flag-gb"
                          src="../assets/blank.gif"
                        />
                        GBP
                      </a>
                    </li>
                  </ul>
                </div>
                <input
                  className="form-control"
                  readOnly={true}
                  type="text"
                  value="loading..."
                />
              </div>
            </div>
          </div>
        </form>
      </div>
      <hr
        className="container"
      />
      <div
        className="history container"
      >
        <h2>
          History
        </h2>
        <div
          className="loader show"
        >
          <div
            className="loader-inner"
          />
        </div>
        <table
          className="table"
        >
          <thead>
            <tr>
              <th>
                Date (YYYY-MM-DD) 
              </th>
              <th>
                Rate (1 
                HKD
                 to 
                USD
                ) 
              </th>
              <th />
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`App.js 2`] = `
<div
  className="App"
>
  <header>
    <nav
      className="navbar navbar-default"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href=""
        >
          Currency Exchange Rate
        </a>
        <div
          className="collapse navbar-collapse"
          id="bs-example-navbar-collapse-1"
        >
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li>
              <a
                href="https://github.com/cklll/exchange-rate"
                target="_blank"
              >
                GitHub
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="MainContainer"
  >
    <div>
      <div
        className="converter container"
      >
        <form>
          <div
            className="container"
          >
            <div
              className="show-grid form-group row"
            >
              <div
                className="col-xs-12"
              >
                <div
                  className="dropdown btn-group btn-group-info"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="dropdown-toggle btn btn-info"
                    disabled={false}
                    id="fromDropwdown"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    type="button"
                  >
                    <span>
                      <img
                        alt="HKD"
                        className="flag flag-hk"
                        src="../assets/blank.gif"
                      />
                      HKD
                    </span>
                     
                    <span
                      className="caret"
                    />
                  </button>
                  <ul
                    aria-labelledby="fromDropwdown"
                    className="dropdown-menu"
                    role="menu"
                  >
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="USD"
                          className="flag flag-us"
                          src="../assets/blank.gif"
                        />
                        USD
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="HKD"
                          className="flag flag-hk"
                          src="../assets/blank.gif"
                        />
                        HKD
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="EUR"
                          className="flag flag-eu"
                          src="../assets/blank.gif"
                        />
                        EUR
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="GBP"
                          className="flag flag-gb"
                          src="../assets/blank.gif"
                        />
                        GBP
                      </a>
                    </li>
                  </ul>
                </div>
                <input
                  className="form-control"
                  defaultValue={1}
                  min="0"
                  onChange={[Function]}
                  type="number"
                />
                <p
                  className="error"
                >
                  Please input a positive number
                </p>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="equal-to col-xs-12"
              >
                <button
                  className="btn btn-primary btn-swap"
                  onClick={[Function]}
                  type="button"
                >
                  â‡…
                </button>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-12"
              >
                <div
                  className="dropdown btn-group btn-group-info"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="dropdown-toggle btn btn-info"
                    disabled={false}
                    id="toDropwdown"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    type="button"
                  >
                    <span>
                      <img
                        alt="USD"
                        className="flag flag-us"
                        src="../assets/blank.gif"
                      />
                      USD
                    </span>
                     
                    <span
                      className="caret"
                    />
                  </button>
                  <ul
                    aria-labelledby="toDropwdown"
                    className="dropdown-menu"
                    role="menu"
                  >
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="USD"
                          className="flag flag-us"
                          src="../assets/blank.gif"
                        />
                        USD
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="HKD"
                          className="flag flag-hk"
                          src="../assets/blank.gif"
                        />
                        HKD
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="EUR"
                          className="flag flag-eu"
                          src="../assets/blank.gif"
                        />
                        EUR
                      </a>
                    </li>
                    <li
                      className=""
                      role="presentation"
                    >
                      <a
                        href="#"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <img
                          alt="GBP"
                          className="flag flag-gb"
                          src="../assets/blank.gif"
                        />
                        GBP
                      </a>
                    </li>
                  </ul>
                </div>
                <input
                  className="form-control"
                  readOnly={true}
                  type="text"
                  value="loading..."
                />
              </div>
            </div>
          </div>
        </form>
      </div>
      <hr
        className="container"
      />
      <div
        className="history container"
      >
        <h2>
          History
        </h2>
        <div
          className="loader show"
        >
          <div
            className="loader-inner"
          />
        </div>
        <table
          className="table"
        >
          <thead>
            <tr>
              <th>
                Date (YYYY-MM-DD) 
              </th>
              <th>
                Rate (1 
                HKD
                 to 
                USD
                ) 
              </th>
              <th />
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
